### DISCUSSION TITLE
Docker, how to backup data

### DISCUSSION THREAD

**Discussion by duchenpaul** [toeverything/AFFiNE#165] [Q&A] [answered]
How can I backup & restore data, I am using docker, how to map the volumn?

**Comment by darkskygit** [comment, marked as answer]
hello, docker images are pure front-end container and do not contain back-end functions, currently all data will be stored in your browser
this week we will launch the local data storage feature, then you will be able to store data locally in real-time, and you can backup and restore it at any time

**Reply by duchenpaul** [reply]
Cheers, how COOL is that!

**Reply by ETWang1991** [reply]
Chrees

**Reply by Jengro777** [reply]
So far I still don't know how to persist data to disk

**Reply by ShortCipher5** [reply]
> So far I still don't know how to persist data to disk

Sorry to hear that. At the time of this message the current version will require you to "sync to disk" to store conent on your local client. This feature is only available when accessing AFFiNE on localhost or over SSL (due to browser API requirements). 

We are currently making some big changes to the current version, which will help create a much more user friendly experience. So please stay tuned for that update.

**Reply by zhouyunjian** [reply]
Chrees，but still waiting！

**Reply by Jengro777** [reply]
It is hoped that the new refactored version will have an optional server-side data storage function. After all, sometimes the way of automatically saving data can improve the user experience and avoid the risk that the data often needs to be moved.

**Reply by ShortCipher5** [reply]
Just to update this discussion. The current Docker version does away with the "sync to disk" feature, the storage functionality has been significantly improved with better compatibility across browsers and devices.

For the server-side storage functionality with multi-user realtime colloboration, you'll want to use the `nightly-server-latest` Docker flag.

**Comment by Jengro777** [comment]
> How can I backup & restore data, I am using docker, how to map the volumn?

It is now ready to store data on the server side
https://github.com/toeverything/AFFiNE/discussions/461

docker pull ghcr.io/toeverything/affine:nightly-latest
docker run -itd  -p 4200:3000 -v {your_path}:/app/data ghcr.io/toeverything/affine:nightly-server-latest
